### PR TITLE
Update to d.js v13 & Subtitle Updates

### LINKED ISSUE #879: "Store channel permissions" is broken
**Discord Bot Maker version (master branch only supports the beta 2.0.x)**
- [x] >2.0.X

**Describe the bug**
Store channel permissions is broken, always shows "mentioned role - undefined - undefined"

**To Reproduce**
1. Add the action
2. Switch target type to member
3. Do anything
4. Save it
5. You will see it is not saving properly and instead shows "mentioned role - undefined - undefined"

**Expected behavior**
It should save the settings properly instead of always saying "mentioned role - undefined - undefined"

**Screenshots**
image
image

### LINKED ISSUE #887: Store Category Info doesn't work anymore
If i want to store the category's channel count nothing happens

### LINKED ISSUE #892: Check if Message Action (can't select anything in the temp variable the box is missing)
**Discord Bot Maker version (master branch only supports the beta 2.0.x)**

- [X ] >2.0.X

**Describe the bug**
There is a bug in the action check if message. There is the problem that I can't select anything in the temp variable in the box because this is simply missing.

**To Reproduce**
1. Go to `Search and Enter check if message `
2. Click on 'check if message'
3. Scroll down to 'Source Message click on Command Message choose temp variable'
4. and then the box to select is missing

**Expected behavior**
That there is the possibility to select the variables

**Screenshots**
https://cdn.discordapp.com/attachments/854808508388999170/998195608475156540/unknown.png

**Error messages**
If there was an error message please put it here. ()

### LINKED ISSUE #905: Check
- [ ] 2.0.X

**Describe the bug**
Check if member source member option shows variables instead of slash parameters when slash parameters are chosen.

**To Reproduce**
1.Add the action on a slash command
2. Change the source Member option

**Screenshots**

image

### PR DESCRIPTION
**Pull request by TheMonDon**
**Please describe the changes this PR makes and why it should be merged:**
I have fixed these two files and have confirmed them to be working inside of DBM.

This PR will be able to resolve multiple issues.
Resolves #905, Resolves #892, Resolves #887, Resolves #879

**Status**

- [X] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [X] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by almeidx — CHANGES_REQUESTED** [review]
Do not rename fields

**Comment by TheMonDon** [comment]
@almeidx can you review this and accept it?

File: actions/set_role_voice_channel_perms_MOD.js [hunk 1]
```diff
@@ -57,12 +57,8 @@ module.exports = {
...
   async action(cache) {
     const data = cache.actions[cache.index];
-    const storage = parseInt(data.vchannel, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const channel = await this.getVoiceChannel(storage, varName, cache);
-    const storage2 = parseInt(data.role, 10);
-    const varName2 = this.evalMessage(data.varName2, cache);
-    const role = await this.getRole(storage2, varName2, cache);
+    const channel = await this.getVoiceChannelFromData(data.storage, data.varName, cache);
+    const role = await this.getRoleFromData(data.storage2, data.varName2, cache);
```
Comment (almeidx) [inline actions/set_role_voice_channel_perms_MOD.js:61]: Use the corresponding HTML elements
The first parameters should be `data.vchannel` and `data.role`

File: actions/set_slowmode_MOD.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ module.exports = {
   },
 
   subtitle(data, presets) {
-    return `${presets.getChannelText(data.storage, data.varName)} : ${data.amount} seconds`;
+    return `${presets.getChannelText(data.channel, data.varName)} : ${data.amount} seconds`;
```
Comment (almeidx) [inline actions/set_slowmode_MOD.js:13]: This field does not exist

File: actions/store_audit_log_list_MOD.js [hunk 3]
```diff
@@ -184,8 +184,7 @@ module.exports = {
       case 0:
         break;
       default: {
-        const varName = this.evalMessage(data.varName, cache);
-        mem = await this.getMember(member - 1, varName, cache);
+        mem = await this.getMemberFromData(data.member - 1, data.varName, cache);
```
Comment (almeidx) [inline actions/store_audit_log_list_MOD.js:187]: Use the corresponding HTML element

File: actions/store_channel_permissions_MOD.js [hunk 4]
```diff
@@ -81,31 +34,18 @@ module.exports = {
...
-    <select id="member" class="round" onchange="glob.memberChange(this, 'varNameContainer3')">
-      ${data.members[isEvent ? 1 : 0]}
-    </select>
-  </div>
-  <div id="varNameContainer3" style="display: none; float: right; width: 60%;">
-    Variable Name:<br>
-    <input id="varName3" class="round" type="text" list="variableList">
-  </div>
-</div><br><br><br>
+  <member-input dropdownLabel="Source Member" selectId="member" variableContainerId="memberVarNameContainer" variableInputId="memberVarName"></member-input>
```
Comment (almeidx) [inline actions/store_channel_permissions_MOD.js:45]: The variableContainerId and variableInputId dont match the old ones

File: actions/store_channel_permissions_MOD.js [hunk 5]
```diff
@@ -81,31 +34,18 @@ module.exports = {
...
-    Source Role:<br>
-    <select id="role" class="round" onchange="glob.roleChange(this, 'varNameContainer2')">
-      ${data.roles[isEvent ? 1 : 0]}
-    </select>
-  </div>
-  <div id="varNameContainer2" style="display: none; float: right; width: 60%;">
-    Variable Name:<br>
-    <input id="varName2" class="round" type="text" list="variableList"><br>
-  </div>
+  <role-input dropdownLabel="Source Role" selectId="role" variableContainerId="roleVarNameContainer" variableInputId="varName2"></role-input>
```
Comment (almeidx) [inline actions/store_channel_permissions_MOD.js:41]: The variableContainerId doesnt match the old ones

File: actions/store_message_params_MOD.js [hunk 6]
```diff
@@ -236,16 +222,18 @@ module.exports = {
         }
         break;
       case 2:
-        result = msg.mentions.users.array().length > 0 ? msg.mentions.users.array()[ParamN - 1] : undefined;
+        result = [...msg.mentions.users.values()].length > 0 ? [...msg.mentions.users.values()][ParamN - 1] : undefined;
```
Comment (almeidx) [inline actions/store_message_params_MOD.js:225]:
```suggestion
        result = msg.mentions.users.at(ParamN - 1);
```

File: actions/store_message_params_MOD.js [hunk 7]
```diff
@@ -236,16 +222,18 @@ module.exports = {
...
         break;
       case 3:
-        result = msg.mentions.members.array().length > 0 ? msg.mentions.members.array()[ParamN - 1] : undefined;
+        result =
+          [...msg.mentions.members.values()].length > 0 ? [...msg.mentions.members.values()][ParamN - 1] : undefined;
```
Comment (almeidx) [inline actions/store_message_params_MOD.js:229]:
```suggestion
        result = msg.mentions.members.at(ParamN - 1);
```

File: actions/store_message_params_MOD.js [hunk 8]
```diff
@@ -236,16 +222,18 @@ module.exports = {
...
         break;
       case 4:
-        result = msg.mentions.roles.array().length > 0 ? msg.mentions.roles.array()[ParamN - 1] : undefined;
+        result = [...msg.mentions.roles.values()].length > 0 ? [...msg.mentions.roles.values()][ParamN - 1] : undefined;
```
Comment (almeidx) [inline actions/store_message_params_MOD.js:232]:
```suggestion
        result = msg.mentions.roles.at(ParamN - 1);
```

File: actions/store_message_params_MOD.js [hunk 9]
```diff
@@ -236,16 +222,18 @@ module.exports = {
...
         break;
       case 5:
-        result = msg.mentions.channels.array().length > 0 ? msg.mentions.channels.array()[ParamN - 1] : undefined;
+        result =
+          [...msg.mentions.channels.values()].length > 0 ? [...msg.mentions.channels.values()][ParamN - 1] : undefined;
```
Comment (almeidx) [inline actions/store_message_params_MOD.js:236]:
```suggestion
        result = msg.mentions.channels.at(ParamN - 1);
```

File: actions/typing_MOD.js [hunk 10]
```diff
@@ -63,9 +62,7 @@ module.exports = {
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const storage = parseInt(data.storage, 10);
-    const varName = this.evalMessage(data.VarName, cache);
-    const channel = await this.getChannel(storage, varName, cache);
+    const channel = await this.getChannelFromData(data.storage, data.varName, cache);
```
Comment (almeidx) [inline actions/typing_MOD.js:65]: Use the corresponding HTML element

File: actions/weather_MOD.js [hunk 11]
```diff
@@ -71,17 +71,19 @@ module.exports = {
 
   html(isEvent, data) {
```
Comment (almeidx) [inline actions/weather_MOD.js:72]:
```suggestion
  html() {
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: actions/parse_from_stored_json_MOD.js [hunk 12]
```diff
@@ -116,7 +122,7 @@ module.exports = {
           outData = Mods.jsonPath(jsonData, `$..${path}`);
         }
 
-        if (DEBUG) console.log(outData);
+        if (data.debugMode) console.log(outData);
```
[the remark this answers is not in the document]
Comment (TheMonDon) [reply actions/parse_from_stored_json_MOD.js:125]: You're right.

File: actions/store_audit_log_list_MOD.js (hunk 3 above)
Comment (TheMonDon) [reply actions/store_audit_log_list_MOD.js:187]: Decided to revert this change,

File: actions/await_reaction_call_action_MOD.js [hunk 13]
```diff
@@ -71,39 +71,40 @@ module.exports = {
...
-        ${data.messages[isEvent ? 1 : 0]}
-      </select>
-    </div>
-    <div id="varNameContainer" style="display: none; float: right; width: 60%;">
-      Variable Name:<br>
-      <input id="varName" class="round" type="text" list="variableList">
-    </div>
-  </div><br><br><br>
-  <div style="width: 567px; margin-top: 8px;">
+    <message-input dropdownLabel="Source Message" selectId="message" variableContainerId="varNameContainer" variableInputId="varName"></message-input>
```
Comment (almeidx) [inline actions/await_reaction_call_action_MOD.js:78]:
```suggestion
    <message-input dropdownLabel="Source Message" selectId="storage" variableContainerId="varNameContainer" variableInputId="varName"></message-input>
```

File: actions/await_reaction_call_action_MOD.js [hunk 14]
```diff
@@ -220,10 +226,7 @@ module.exports = {
   async action(cache) {
     const data = cache.actions[cache.index];
     const { Actions } = this.getDBM();
-
-    const messageVariable = parseInt(data.storage, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const msg = await this.getMessage(messageVariable, varName, cache);
+    const msg = await this.getMessageFromData(data.message, data.varName, cache);
```
Comment (almeidx) [inline actions/await_reaction_call_action_MOD.js:229]:
```suggestion
    const msg = await this.getMessageFromData(data.storage, data.varName, cache);
```

File: actions/check_dbl_voted_MOD.js [hunk 15]
```diff
@@ -106,17 +99,15 @@ module.exports = {
...
     glob.onChangeFalse(document.getElementById('iffalse'));
   },
 
   async action(cache) {
     const data = cache.actions[cache.index];
     const apitoken = this.evalMessage(data.apitoken, cache);
-    const type = parseInt(data.member, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const member = await this.getMember(type, varName, cache);
+    const member = await this.getMemberFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/check_dbl_voted_MOD.js:110]:
```suggestion
    const member = await this.getMemberFromData(data.member, data.varName, cache);
```

File: actions/check_if_member_MOD.js [hunk 16]
```diff
@@ -131,9 +130,7 @@ module.exports = {
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const type = parseInt(data.member, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const member = await this.getMember(type, varName, cache);
+    const member = await this.getMemberFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/check_if_member_MOD.js:133]:
```suggestion
    const member = await this.getMemberFromData(data.member, data.varName, cache);
```

File: actions/check_if_user_reacted_MOD.js [hunk 17]
```diff
@@ -120,10 +105,7 @@ module.exports = {
 
   async action(cache) {
     const data = cache.actions[cache.index];
-
-    const type = parseInt(data.member, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const member = await this.getMember(type, varName, cache);
+    const member = await this.getMemberFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/check_if_user_reacted_MOD.js:108]:
```suggestion
    const member = await this.getMemberFromData(data.member, data.varName, cache);
```

File: actions/check_role_permissions_MOD.js [hunk 18]
```diff
@@ -105,16 +90,14 @@ module.exports = {
...
     glob.onChangeTrue(document.getElementById('iftrue'));
     glob.onChangeFalse(document.getElementById('iffalse'));
   },
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const storage = parseInt(data.role, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const role = await this.getRole(storage, varName, cache);
+    const role = await this.getRoleFromData(data.storage, data.varName, cache);
```
Comment (almeidx) [inline actions/check_role_permissions_MOD.js:100]:
```suggestion
    const role = await this.getRoleFromData(data.role, data.varName, cache);
```

File: actions/clone_channel_MOD.js [hunk 19]
```diff
@@ -9,17 +9,8 @@ module.exports = {
...
-      'Mentioned Channel',
-      'Default Channel',
-      'Temp Variable',
-      'Server Variable',
-      'Global Variable',
-    ];
-    const index = parseInt(data.storage, 10);
-    return index < 3 ? `Clone Channel : ${names[index]}` : `Clone Channel : ${names[index]} - ${data.varName}`;
+  subtitle(data, presets) {
+    return `Clone Channel: ${presets.getChannelText(data.channel, data.varName)}`;
```
Comment (almeidx) [inline actions/clone_channel_MOD.js:13]:
```suggestion
    return `Clone Channel: ${presets.getChannelText(data.storage, data.varName)}`;
```

File: actions/control_server_prefix_MOD.js [hunk 20]
```diff
@@ -60,11 +52,9 @@ module.exports = {
     const fs = require('fs');
     const path = require('path');
     const data = cache.actions[cache.index];
-    const type = parseInt(data.server, 10);
     const { Actions } = this.getDBM();
 
-    const varName = this.evalMessage(data.varName, cache);
-    const server = await this.getServer(type, varName, cache);
+    const server = await this.getServerFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/control_server_prefix_MOD.js:57]:
```suggestion
    const server = await this.getServerFromData(data.server, data.varName, cache);
```

File: actions/delete_member_data_MOD.js [hunk 21]
```diff
@@ -38,16 +29,11 @@ module.exports = {
...
-    glob.memberChange(document.getElementById('member'), 'varNameContainer');
-  },
+  init() {},
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const type = parseInt(data.member, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const member = await this.getMember(type, varName, cache);
+    const member = await this.getMemberFromData(data.type, data.varName, data.cache);
```
Comment (almeidx) [inline actions/delete_member_data_MOD.js:36]:
```suggestion
    const member = await this.getMemberFromData(data.member, data.varName, data.cache);
```

File: actions/delete_server_data_MOD.js [hunk 22]
```diff
@@ -45,9 +36,7 @@ module.exports = {
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const type = parseInt(data.server, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const server = await this.getServer(type, varName, cache);
+    const server = await this.getServerFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/delete_server_data_MOD.js:39]:
```suggestion
    const server = await this.getServerFromData(data.server, data.varName, cache);
```

File: actions/leave_server_MOD.js [hunk 23]
```diff
@@ -15,32 +15,15 @@ module.exports = {
...
-    glob.serverChange(document.getElementById('server'), 'varNameContainer');
-  },
+  init() {},
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const type = parseInt(data.server, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const server = await this.getServer(type, varName, cache);
+    const server = await this.getServerFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/leave_server_MOD.js:26]:
```suggestion
    const server = await this.getServerFromData(data.server, data.varName, cache);
```

File: actions/remove_reaction_MOD.js [hunk 24]
```diff
@@ -29,26 +29,17 @@ module.exports = {
...
-  </div>
-  <div id="varNameContainer2" style="display: none; float: right; width: 60%;">
-    Variable Name:<br>
-    <input id="varName2" class="round" type="text" list="variableList"><br>
-  </div>
-</div>`;
+</div>
+<br><br><br><br>
+
+<member-input dropdownLabel="Source Member" selectId="member" variableContainerId="varNameContainer" variableInputId="varName"></member-input>
```
Comment (almeidx) [inline actions/remove_reaction_MOD.js:35]:
```suggestion
<member-input dropdownLabel="Source Member" selectId="member" variableContainerId="varNameContainer2" variableInputId="varName2"></member-input>
```

File: actions/set_afk_channel_MOD.js [hunk 25]
```diff
@@ -23,83 +23,21 @@ module.exports = {
...
 
   async action(cache) {
     const data = cache.actions[cache.index];
-    const type = parseInt(data.server, 10);
-    const afkchannel = parseInt(data.afkchannel, 10);
-    const varName2 = this.evalMessage(data.varNameChannel, cache);
-    const varName = this.evalMessage(data.varName, cache);
-    const server = await this.getServer(type, varName, cache);
-    const channel = await this.getVoiceChannel(afkchannel, varName2, cache);
+    const server = await this.getServerFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/set_afk_channel_MOD.js:39]:
```suggestion
    const server = await this.getServerFromData(data.server, data.varName, cache);
```

File: actions/set_afk_channel_MOD.js [hunk 26]
```diff
@@ -23,83 +23,21 @@ module.exports = {
...
+    const channel = await this.getVoiceChannelFromData(data.afkchannel, data.varName2, cache);
```
Comment (almeidx) [inline actions/set_afk_channel_MOD.js:40]:
```suggestion
    const channel = await this.getVoiceChannelFromData(data.afkchannel, data.varNameChannel, cache);
```

File: actions/set_member_voice_channel_perms_MOD.js [hunk 27]
```diff
@@ -61,13 +61,8 @@ module.exports = {
...
   async action(cache) {
     const data = cache.actions[cache.index];
-    const storage = parseInt(data.vchannel, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const channel = await this.getVoiceChannel(storage, varName, cache);
-
-    const storage2 = parseInt(data.member, 10);
-    const varName2 = this.evalMessage(data.varName2, cache);
-    const member = await this.getMember(storage2, varName2, cache);
+    const channel = await this.getVoiceChannelFromData(data.storage, data.varName, cache);
```
Comment (almeidx) [inline actions/set_member_voice_channel_perms_MOD.js:64]:
```suggestion
    const channel = await this.getVoiceChannelFromData(data.vchannel, data.varName, cache);
```

File: actions/set_member_voice_channel_perms_MOD.js [hunk 28]
```diff
@@ -61,13 +61,8 @@ module.exports = {
...
+    const member = await this.getMemberFromData(data.storage2, data.varName2, cache);
```
Comment (almeidx) [inline actions/set_member_voice_channel_perms_MOD.js:65]:
```suggestion
    const member = await this.getMemberFromData(data.member, data.varName2, cache);
```

File: actions/set_permissions_MOD.js [hunk 29]
```diff
@@ -68,20 +59,13 @@ module.exports = {
...
     const data = cache.actions[cache.index];
-    const storage = parseInt(data.storage, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const storage2 = parseInt(data.storage2, 10);
-    const varName2 = this.evalMessage(data.varName2, cache);
-
-    const role = await this.getRole(storage, varName, cache);
-    let permissions = this.getVariable(storage2, varName2, cache);
+    const role = await this.getRoleFromData(data.storage, data.varName, cache);
+    let permissions = this.getVariableFromData(data.storage2, data.varName2, cache);
```
Comment (almeidx) [inline actions/set_permissions_MOD.js:68]: I don't believe this function exists

File: actions/set_server_owner_MOD.js [hunk 30]
```diff
@@ -18,56 +18,28 @@ module.exports = {
...
-
-    const type = parseInt(data.server, 10);
-    const varName = this.evalMessage(data.varName, cache);
-    const server = await this.getServer(type, varName, cache);
-
-    const member = parseInt(data.member, 10);
-    const varName2 = this.evalMessage(data.varName2, cache);
-    const mem = await this.getMember(member, varName2, cache);
-
+    const server = await this.getServerFromData(data.type, data.varName, cache);
```
Comment (almeidx) [inline actions/set_server_owner_MOD.js:41]:
```suggestion
    const server = await this.getServerFromData(data.server, data.varName, cache);
```

File: actions/store_category_info_MOD.js [hunk 31]
```diff
@@ -187,22 +187,26 @@ module.exports = {
...
+        result = [...targetCategory.children.values()]; // Category Channel List
         break;
       case 7:
         result = targetCategory.children.size; // Category Channel Count
         break;
       case 8:
-        result = targetCategory.children.filter((c) => ['text', 'news', 'store'].includes(c.type)).array(); // Category Text Channel List
+        result = targetCategory.children
+          .filter((c) => ['GUILD_TEXT', 'GUILD_NEWS', 'GUILD_STORE'].includes(c.type))
+          .array(); // Category Text Channel List
```
Comment (almeidx) [inline actions/store_category_info_MOD.js:198]:
```suggestion
        result = [...targetCategory.children
          .filter((c) => ['GUILD_TEXT', 'GUILD_NEWS', 'GUILD_STORE'].includes(c.type))
          .values()]; // Category Text Channel List
```

File: actions/store_category_info_MOD.js [hunk 32]
```diff
@@ -187,22 +187,26 @@ module.exports = {
...
         break;
       case 9:
-        result = targetCategory.children.filter((c) => ['text', 'news', 'store'].includes(c.type)).size; // Category Text Channel Count
+        result = targetCategory.children.filter((c) =>
+          ['GUILD_TEXT', 'GUILD_NEWS', 'GUILD_STORE'].includes(c.type),
+        ).size; // Category Text Channel Count
         break;
       case 10:
-        result = targetCategory.children.filter((c) => c.type === 'voice').array(); // Category Voice Channel List
+        result = targetCategory.children.filter((c) => c.type === 'GUILD_VOICE').array(); // Category Voice Channel List
```
Comment (almeidx) [inline actions/store_category_info_MOD.js:206]:
```suggestion
        result = [...targetCategory.children.filter((c) => c.type === 'GUILD_VOICE').values()]; // Category Voice Channel List
```

File: actions/store_channel_permissions_MOD.js [hunk 33]
```diff
@@ -10,50 +10,10 @@ module.exports = {
...
-      'Deafen Members',
-      'Move Members',
-      'Use Voice Activity',
-      'All Permissions',
-    ];
-    return `${roles[data.role]} - ${perm[data.permission]} - ${index[data.state]} ${
-      !data.reason ? '' : `with Reason: <i>${data.reason}<i>`
+  subtitle(data, presets) {
+    const target = parseInt(data.target, 10);
+    return `${presets.getChannelText(data.channel, data.channelVarNameContainer)} - ${
```
Comment (almeidx) [inline actions/store_channel_permissions_MOD.js:15]:
```suggestion
    return `${presets.getChannelText(data.storage, data.varName)} - ${
```

File: actions/store_channel_permissions_MOD.js [hunk 34]
```diff
@@ -81,31 +34,18 @@ module.exports = {
...
-    <select id="member" class="round" onchange="glob.memberChange(this, 'varNameContainer3')">
-      ${data.members[isEvent ? 1 : 0]}
-    </select>
-  </div>
-  <div id="varNameContainer3" style="display: none; float: right; width: 60%;">
-    Variable Name:<br>
-    <input id="varName3" class="round" type="text" list="variableList">
-  </div>
-</div><br><br><br>
+  <member-input dropdownLabel="Source Member" selectId="member" variableContainerId="arNameContainer3" variableInputId="memberVarName"></member-input>
```
Comment (almeidx) [inline actions/store_channel_permissions_MOD.js:45]:
```suggestion
  <member-input dropdownLabel="Source Member" selectId="member" variableContainerId="varNameContainer3" variableInputId="varName3"></member-input>
```

File: actions/store_channel_permissions_MOD.js [hunk 35]
```diff
@@ -81,31 +34,18 @@ module.exports = {
...
-    Source Role:<br>
-    <select id="role" class="round" onchange="glob.roleChange(this, 'varNameContainer2')">
-      ${data.roles[isEvent ? 1 : 0]}
-    </select>
-  </div>
-  <div id="varNameContainer2" style="display: none; float: right; width: 60%;">
-    Variable Name:<br>
-    <input id="varName2" class="round" type="text" list="variableList"><br>
-  </div>
+  <role-input dropdownLabel="Source Role" selectId="role" variableContainerId="arNameContainer2" variableInputId="varName2"></role-input>
```
Comment (almeidx) [inline actions/store_channel_permissions_MOD.js:41]:
```suggestion
  <role-input dropdownLabel="Source Role" selectId="role" variableContainerId="varNameContainer2" variableInputId="varName2"></role-input>
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request merged**